nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/upgrade/P2Upgrade_1_1.java

```diff
@@ -0,0 +1,137 @@
+import org.sonatype.nexus.orient.DatabaseInstance;
+import org.sonatype.nexus.orient.DatabaseInstanceNames;
+import org.sonatype.nexus.orient.DatabaseUpgradeSupport;
+import org.sonatype.nexus.orient.OClassNameBuilder;
+import org.sonatype.nexus.orient.OIndexNameBuilder;
+
+import com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx;
+import com.orientechnologies.orient.core.db.record.OIdentifiable;
+import com.orientechnologies.orient.core.index.OIndex;
+import com.orientechnologies.orient.core.metadata.schema.OSchemaProxy;
+import com.orientechnologies.orient.core.record.impl.ODocument;
+import com.orientechnologies.orient.core.sql.OCommandSQL;
+import com.orientechnologies.orient.core.sql.query.OSQLSynchQuery;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Upgrade step to set {@code name} for p2 assets.
+ *
+ * @since 1.0.0
+ */
+@Named
+@Singleton
+@Upgrades(model = P2Model.NAME, from = "1.0", to = "1.1")
+@DependsOn(model = DatabaseInstanceNames.COMPONENT, version = "1.14", checkpoint = true)
+@DependsOn(model = DatabaseInstanceNames.CONFIG, version = "1.8", checkpoint = true)
+public class P2Upgrade_1_1
+    extends DatabaseUpgradeSupport
+{
+  private static final String SELECT_P2_REPOSITORIES =
+      "select from repository where recipe_name = 'p2-proxy'";
+
+  private static final String REMOVE_UNNECESSARY_SLASH_FROM_ASSET_NAME =
+      "update asset set name = name.subString(1) where bucket = ? and name like '/%'";
+
+  private static final String P_REPOSITORY_NAME = "repository_name";
+
+  private static final String C_BROWSE_NODE = "browse_node";
+
+  private static final String BROWSE_NODE_CLASS = new OClassNameBuilder().type(C_BROWSE_NODE).build();
+
+  private static final String DELETE_BROWSE_NODE_FROM_REPOSITORIES = String
+      .format("delete from %s where repository_name in ?", BROWSE_NODE_CLASS);
+
+  private static final String I_REPOSITORY_NAME = new OIndexNameBuilder()
+      .type("bucket")
+      .property(P_REPOSITORY_NAME)
+      .build();
+
+  private final Provider<DatabaseInstance> configDatabaseInstance;
+
+  private final Provider<DatabaseInstance> componentDatabaseInstance;
+
+  @Inject
+  public P2Upgrade_1_1(
+      @Named(DatabaseInstanceNames.CONFIG) final Provider<DatabaseInstance> configDatabaseInstance,
+      @Named(DatabaseInstanceNames.COMPONENT) final Provider<DatabaseInstance> componentDatabaseInstance)
+  {
+    this.configDatabaseInstance = checkNotNull(configDatabaseInstance);
+    this.componentDatabaseInstance = checkNotNull(componentDatabaseInstance);
+  }
+
+  @Override
+  public void apply() {
+    if (hasSchemaClass(configDatabaseInstance, "repository") &&
+        hasSchemaClass(componentDatabaseInstance, "asset")) {
+      updateP2Repositories();
+    }
+  }
+
+  private void updateP2Repositories() {
+    List<String> p2RepositoryNames;
+    try (ODatabaseDocumentTx db = configDatabaseInstance.get().connect()) {
```

Comment:
Question: any reason why not to use the `withDatabaseAndClass` of the `DatabaseUpgradeSupport` class that you are extending? Instead of setting up the connection manually here again.

Reply:
Done